main.js

```diff
@@ -0,0 +1,289 @@
+
+const outDir = 'html'; // TODO canonicalise?
+
+fse.mkdirpSync(outDir);
+fse.emptyDirSync(outDir);
+
+compileTemplate = template => Handlebars.compile(template.toString('UTF-8'));
+
+/**
+ * Wraps HTML documents found in dirname with wrapperTemplate and writes them to outDir
+ * @param {string} dirname The directory to read HTML content from
+ * @param {Handlebars.TemplateDelegate} wrapperTemplate Handlebars template to apply to the content
+ * @param {Object} globalContext Context to use for every page
+ * @returns {Promise} no resolve value, wraps around a Promise.all of all of the files
+ */
+function regularDir(dirname, wrapperTemplate, globalContext) {
+    return fse.readdir(dirname).then(listing => {
+        /**
+         * {String[]} listing: filenames in regular/
+         */
+        let promises = listing //.filter(fn => /.*\.html/.test(fn)) // Filter to only HTML files
+            .map(filename => 
+                fse.readFile(path.join(dirname, filename))// open each file
+                .then(content => {
+                    /**
+                     * Takes content and applies the wraper template to it
+                     * content begins with a YAML header.
+                     * {Buffer} content: contents of `filename`
+                     */
+
+                    let matter = docmatter(content.toString('UTF-8'));
+
+                    return wrapperTemplate(Object.assign({ // Merge this object with the properties in context.yaml
+                        body: matter.body
+                    }, globalContext, yaml.safeLoad(matter.header)))
+                })
+                .then(html => fse.writeFile(path.join(outDir, filename), html))
+            );
+        return Promise.all(promises)
+    }).catch(err => {
+        console.log(`Error processing regular directory ${dirname}/ - ${err}`);
+    });
+}
+
+/**
+ * Renders Markdown server READMEs found in dirname, wrap with template and write to outDir/dirname/
+ * @param {string} dirname name to find Markdown source files
+ * @param {Handbars.TemplateDelegate} serverTemplate Handlebars template to make the page
+ * @param {Object} globalContext Misc context (navbar, servers, etc)
+ * @returns {Promise} wraps around a Promise.all for each file in `dirname`
+ */
+function servers(dirname, serverTemplate, globalContext) {
+    return fse.readdir(dirname).then(listing => {
+        /**
+         * {String[]} listing: filenames in servers/
+         */
+        let promises = listing.map(filename => 
+                fse.readFile(path.join(dirname,filename))
+                .then(content => {
+                    /**
+                     * {Buffer} content: contents of `filename`
+                     */
+                    let matter = docmatter(content.toString('UTF-8'));
+
+                    return serverTemplate(
+                        Object.assign({
+                            body: Markdown.render(matter.body)
```

Comment:
`Markdown.render(matter.body)` on L76 will fail if the input file is empty with `Error: Input data should be a String`. Unsure if this is caused by the body being undefined in the output of `docmatter` or an internal `Markdown` issue.

This issue affects _all_ markdown parsing in this script, not just this line.